File: Sorts/Sorts.java
public class Sorts{
	private int[] ary;

	public Sorts(int[] ary){
		this.ary = ary;
	}

	public int[] bubbleSort(){
		int a = ary.length;
		while(a > 0){
			for(int j = 0; j < a - 1; j++){
				if(ary[j] > ary[j + 1]){
					int temp = ary[j];
					ary[j] = ary[j + 1];
					ary[j + 1] = temp;
				}
			}
			a--;
		}
		return ary;
	}

	public int[] selectionSort(){
		int a = 0;
		while(a < ary.length){
			int min = ary[a];
			int index = a;
			for(int j = a; j < ary.length; j++){
				if(ary[j] < min){
					min = ary[j];
					index = j;
				}
			}
			ary[index] = ary[a];
			ary[a] = min;
			a++;
		}
		return ary;
	}

	public int[] insertionSort(){
		int a = 0;
		while(a < ary.length){
			int j = a;
			int temp = ary[a];
			while(j > 0 && ary[j - 1] > temp){
				ary[j] = ary[j - 1];
				j--;
			}
			ary[j] = temp;
			a++;
		}
		return ary;
	}

	public int[] quickSort(){
		return quickSortHelp(0, ary.length - 1);
	}

	private int[] quickSortHelp(int left, int right){
		if(left == right || left > right){
			return ary;
		}
		else if(left + 1 == right){
			if(ary[left] > ary[right]){
				int temp = ary[left];
				ary[left] = ary[right];
				ary[right] = temp;
				return ary;
			}
		}
		else{
			int pivot = right;
			int l = left;
			int r = right - 1;
			while(l < r){
				if(ary[l] <= ary[pivot] && ary[r] > ary[pivot]){
					l++;
					r--;
				}
				else if(ary[l] <= ary[pivot] && ary[r] <= ary[pivot]){
					l++;
				}
				else if(ary[l] > ary[pivot] && ary[r] > ary[pivot]){
					r--;
				}
				else if(ary[l] > ary[pivot] && ary[r] <= ary[pivot]){
					int temp = ary[l];
					ary[l] = ary[r];
					ary[r] = temp;
					l++;
					r--;
				}
			}
			if(ary[l] < ary[pivot]){
				l++;
			}
			int temp = ary[l];
			ary[l] = ary[pivot];
			ary[pivot] = temp;
			quickSortHelp(left, l - 1);
			quickSortHelp(l + 1, right);
		}
		return ary;
	}

	public int[] mergeSort(){
		if(ary.length <= 1){
			return ary
		}
		else{
			ary = mergeSortHelp(ary)
		}
	}

	private int[] mergeSortHelp(int[] ary){
		if(ary.length <= 1){
			return ary
		}
		else{
			int[] ary = int[8]
		}
	}

	public String toString(){
		String s = "[";
		for(int i = 0; i < ary.length; i++){
			if(i == ary.length - 1){
				s += ary[i] + "]";
			}
			else{
				s += ary[i] + ", ";
			}
		}
		return s;
	}

	public static void main(String[] args){
		int[] ary0 = {6, 5, 3, 1, 8, 7, 2, 4};
		Sorts s = new Sorts(ary0);
		try{
			if(args[0].equals("BubbleSort")){
				System.out.println("Java Bubble Sort");
				System.out.println(s.toString());
				s.bubbleSort();
				System.out.println(s.toString());
			}
			else if(args[0].equals("SelectionSort")){
				System.out.println("Java Selection Sort");
				System.out.println(s.toString());
				s.selectionSort();
				System.out.println(s.toString());
			}
			else if(args[0].equals("InsertionSort")){
				System.out.println("Java Insertion Sort");
				System.out.println(s.toString());
				s.insertionSort();
				System.out.println(s.toString());
			}
			else if(args[0].equals("QuickSort")){
				System.out.println("Java Quick Sort");
				System.out.println(s.toString());
				s.quickSort();
				System.out.println(s.toString());
			}
		}
		catch (IndexOutOfBoundsException e){
			System.out.println("Please indicate which sort you would like to run by typing 'java Sorts <X>Sort', where X is the name of the sort");
		}
	}
}

File: Sorts/Sorts.py
import random
import sys
class Heap:

	def __init__(self):
		_A = []

	def add(self, i):
		_A.append
class Sorts:

	def __init__(self, ary):
		self._ary = ary

	def bogoSort(self):
		sorted = False
		if len(self._ary) <= 1:
			return self._ary
		while not sorted:
			i = 0
			while self._ary[i] <= self._ary[i + 1]:
				if i == len(self._ary) - 2:
					if self._ary[i] <= self._ary[i + 1]:
						return self._ary
				i += 1
			random.shuffle(self._ary)

	def bubbleSort(self):
		a = len(self._ary)
		while a > 0:
			for j in range(a - 1):
				if self._ary[j] > self._ary[j + 1]:
					temp = self._ary[j]
					self._ary[j] = self._ary[j + 1]
					self._ary[j + 1] = temp
			a -= 1
		return self._ary

	def selectionSort(self):
		a = 0
		while a < len(self._ary):
			min = self._ary[a]
			index = a
			for j in range(a, len(self._ary)):
				if self._ary[j] < min:
					min = self._ary[j]
					index = j
			self._ary[index] = self._ary[a]
			self._ary[a] = min
			a += 1
		return self._ary

	def insertionSort(self):
		a = 0
		while a < len(self._ary):
			j = a
			temp = self._ary[a]
			while j > 0 and self._ary[j - 1] > temp:
				self._ary[j] = self._ary[j - 1]
				j -= 1
			self._ary[j] = temp
			a += 1
		return self._ary

	def quickSort(self):
		return self._quickSortHelp(0, len(self._ary) - 1)

	def _quickSortHelp(self, left, right):
		if left == right or left > right:
			return self._ary
		elif left + 1 == right:
			if self._ary[left] > self._ary[right]:
				temp = self._ary[left]
				self._ary[left] = self._ary[right]
				self._ary[right] = temp
				return self._ary
		else:
			pivot = right
			l = left
			r = right - 1
			while l < r:
				if self._ary[l] <= self._ary[pivot] and self._ary[r] > self._ary[pivot]:
					l += 1
					r -= 1
				elif self._ary[l] <= self._ary[pivot] and self._ary[r] <= self._ary[pivot]:
					l += 1
				elif self._ary[l] > self._ary[pivot] and self._ary[r] > self._ary[pivot]:
					r -= 1
				elif self._ary[l] > self._ary[pivot] and self._ary[r] <= self._ary[pivot]:
					temp = self._ary[l]
					self._ary[l] = self._ary[r]
					self._ary[r] = temp
					l += 1
					r -= 1
			if self._ary[l] < self._ary[pivot]:
				l += 1
			temp = self._ary[l]
			self._ary[l] = self._ary[pivot]
			self._ary[pivot] = temp
			self._quickSortHelp(left, l - 1)
			self._quickSortHelp(l + 1, right)

	def mergeSort(self):
		if len(self._ary) == 0 or len(self._ary) == 1:
			return self._ary
		else:
			self._ary = self._mergeSortHelp(self._ary)

	def _mergeSortHelp(self, ary):
		if len(ary) <= 1:
			return ary
		else:
			ary1 = []
			ary2 = []
			for i in range(len(ary)):
				if i < len(ary) // 2:
					ary1.append(ary[i])
				else:
					ary2.append(ary[i])
			ary1 = self._mergeSortHelp(ary1)
			ary2 = self._mergeSortHelp(ary2)
			return self._merge(ary1, ary2)

	def _merge(self, ary1, ary2):
		l = 0
		r = 0
		i = 0
		merged = []
		while l < len(ary1) and r < len(ary2):
			if ary1[l] <= ary2[r]:
				merged.append(ary1[l])
				l += 1
				i += 1
			elif ary2[r] <= ary1[l]:
				merged.append(ary2[r])
				r += 1
				i += 1
		while l < len(ary1):
			merged.append(ary1[l])
			i += 1
			l += 1
		while r < len(ary2):
			merged.append(ary2[r])
			i += 1
			r += 1
		return merged

	def radixSort(self):
		raise NotImplementedError("This operation is currently unsupported")

	def toString(self):
		s = "["
		for i in range(len(self._ary)):
			if i == len(self._ary) - 1:
				s += str(self._ary[i]) + "]"
			else:
				s += str(self._ary[i]) + ", "
		return s

def main():
	ary0 = [6, 5, 3, 1, 8, 7, 2, 4]
	"""ary0 = []
	for i in range(int(sys.argv[2])):
		ary0.append(random.randint(0, i))"""
	s = Sorts(ary0)
	try:
		if sys.argv[1] == "BogoSort":
			print("Python Bogo Sort")
			print(s.toString())
			s.bogoSort()
			print(s.toString())
			return
		elif sys.argv[1] == "BubbleSort":
			print("Python Bubble Sort")
			print(s.toString())
			s.bubbleSort()
			print(s.toString())
			return
		elif sys.argv[1] == "SelectionSort":
			print("Python Selection Sort")
			print(s.toString())
			s.selectionSort()
			print(s.toString())
			return
		elif sys.argv[1] == "InsertionSort":
			print("Python Insertion Sort")
			print(s.toString())
			s.insertionSort()
			print(s.toString())
			return
		elif sys.argv[1] == "QuickSort":
			print("Python Quick Sort")
			print(s.toString())
			s.quickSort()
			print(s.toString())
			return
		elif sys.argv[1] == "MergeSort":
			print("Python Merge Sort")
			print(s.toString())
			s.mergeSort()
			print(s.toString())
		else:
			raise IndexError()
	except IndexError:
		print("Please indicate which sort you would like to run by typing 'python3 Sorts.py <X>Sort', where X is the name of the sort")

main()